lib/manager/terraform/lockfile/index.spec.ts

```diff
@@ -0,0 +1,411 @@
+    fs.readLocalFile.mockResolvedValueOnce(validLockfile as any);
+
+    mockHash.mockResolvedValueOnce([
+      'h1:lDsKRxDRXPEzA4AxkK4t+lJd3IQIP2UoaplJGjQSp2s=',
+      'h1:6zB2hX7YIOW26OrKsLJn0uLMnjqbPNxcz9RhlWEuuSY=',
+    ]);
+
+    const localConfig: UpdateArtifactsConfig = {
+      updateType: 'minor',
+      newVersion: '2.56.0',
+      newValue: '~> 2.50',
+      ...config,
+    };
+
+    process.env.RENOVATE_TERRAFORM_LOCK_FILE = 'test';
+
+    const result = await updateArtifacts({
+      packageFileName: 'main.tf',
+      updatedDeps: ['azurerm'],
+      newPackageFileContent: '',
+      config: localConfig,
+    });
+    expect(result).not.toBeNull();
+    expect(result).toBeArrayOfSize(1);
+    expect(result[0].file).not.toBeNull();
+    expect(result[0].file).toMatchSnapshot();
+
+    expect(mockHash.mock.calls).toBeArrayOfSize(1);
+    expect(mockHash.mock.calls).toMatchSnapshot();
+  });
+
+  it('update single dependency with range constraint and major update', async () => {
+    fs.readLocalFile.mockResolvedValueOnce(validLockfile as any);
+
+    mockHash.mockResolvedValueOnce([
+      'h1:lDsKRxDRXPEzA4AxkK4t+lJd3IQIP2UoaplJGjQSp2s=',
+      'h1:6zB2hX7YIOW26OrKsLJn0uLMnjqbPNxcz9RhlWEuuSY=',
+    ]);
+
+    const localConfig: UpdateArtifactsConfig = {
+      updateType: 'major',
+      newVersion: '3.1.0',
+      newValue: '~> 3.0',
+      ...config,
+    };
+
+    process.env.RENOVATE_TERRAFORM_LOCK_FILE = 'test';
+
+    const result = await updateArtifacts({
+      packageFileName: 'main.tf',
+      updatedDeps: ['random'],
+      newPackageFileContent: '',
+      config: localConfig,
+    });
+    expect(result).not.toBeNull();
+    expect(result).toBeArrayOfSize(1);
+    expect(result[0].file).not.toBeNull();
+    expect(result[0].file).toMatchSnapshot();
+
+    expect(mockHash.mock.calls).toBeArrayOfSize(1);
+    expect(mockHash.mock.calls).toMatchSnapshot();
+  });
+
+  it('do full lock file maintenance', async () => {
+    fs.readLocalFile.mockResolvedValueOnce(validLockfile as any);
+
+    mockGetPkgReleases
+      .mockResolvedValueOnce(
+        new Promise((resolve) =>
+          resolve({
+            // aws
+            releases: [
+              {
+                version: '2.30.0',
+              },
+              {
+                version: '3.0.0',
+              },
+              {
+                version: '3.36.0',
+              },
+            ],
+          })
+        )
+      )
+      .mockResolvedValueOnce(
+        new Promise((resolve) =>
+          resolve({
+            // azurerm
+            releases: [
+              {
+                version: '2.50.0',
+              },
+              {
+                version: '2.55.0',
+              },
+              {
+                version: '2.56.0',
+              },
+            ],
+          })
+        )
+      )
+      .mockResolvedValueOnce(
+        new Promise((resolve) =>
```

Comment:
Why using a promise here? 🤔 `mockResolved` will Auto relate a promise.

Reply:
If I am not setting it up like that, the order of the results will become randomized and therefore the `matchSnapshot` function will fail from time to time.

Reply:
Nope, jest makes sure `mockResolvedValueOnce` calls will be used in reverse order. So last mock will be used first. 😉 

See my commit, it works for me.